test/integration/targets/aws_secret/tasks/main.yaml

```diff
@@ -0,0 +1,273 @@
+  - name: dummy lambda for testing
+    lambda:
+      <<: *aws_connection_info
+      name: "{{ resource_prefix }}-hello-world"
+      state: present
+      zip_file: "/root/ansible_testing/hello_world.zip"
+      runtime: 'python2.7'
+      role: "arn:aws:iam::{{ test_caller_facts.account }}:role/{{ resource_prefix }}-test-secrets-manager-role"
+      handler: 'hello_world.lambda_handler'
+
+  - name: lambda policy for secrets manager
+    lambda_policy:
+      <<: *aws_connection_info
+      state: present
+      function_name: "{{ resource_prefix }}-hello-world"
+      statement_id: LambdaSecretsManagerTestPolicy
+      action: 'lambda:InvokeFunction'
+      principal: secretsmanager.amazonaws.com
+
+  # ============================================================
+  # Module parameter testing
+  # ============================================================
+  - name: test with no parameters
+    aws_secret:
+    register: result
+    ignore_errors: true
+    check_mode: true
+
+  - name: assert failure when called with no parameters
+    assert:
+      that:
+         - result.failed
+         - 'result.msg.startswith("missing required arguments:")'
+
+  - name: test with missing parameters
+    aws_secret:
+      name: 'test_secret_string'
+    register: result
+    ignore_errors: true
+    check_mode: true
+
+  - name: assert failure when called with missing parameters
+    assert:
+      that:
+         - result.failed
+         - 'result.msg.startswith("missing required arguments:")'
+
+  - name: test with missing parameters
+    aws_secret:
+      name: 'test_secret_string'
+      state: present
+    register: result
+    ignore_errors: true
+    check_mode: true
+
+  - name: assert failure when called with missing parameters
+    assert:
+      that:
+         - result.failed
+         - 'result.msg.startswith("missing required arguments:")'
+
+  - name: test with missing parameters
+    aws_secret:
+      name: 'test_secret_string'
+      state: present
+      secret_type: 'string'
+    register: result
+    ignore_errors: true
+    check_mode: true
+
+  - name: assert failure when called with missing parameters
+    assert:
+      that:
+         - result.failed
+         - 'result.msg.startswith("missing required arguments:")'
+
+  # ============================================================
+  # Creation/Deletion testing
+  # ============================================================
+  - name: add secret to AWS Secrets Manager
+    aws_secret:
+      <<: *aws_connection_info
+      name: "{{ resource_prefix }}-test-secret-string"
+      state: present
+      secret_type: 'string'
+      secret: "{{ super_secret_string }}"
+    register: result
+
+  - name: assert correct keys are returned
+    assert:
+      that:
+        - result.changed
+        - result.secret_arn is not none
+        - result.version_id is not none
+
+  - name: pausing for secret creation
+    pause:
+      seconds: 30
+
+  - name: no changes to secret
```

Comment:
For some reason this task is failing for me with ResourceExistsException right now.